docs/content/concepts/assets/asset-auto-execution.mdx

```diff
@@ -0,0 +1,169 @@
+---
+title: Auto-Materializing Assets | Dagster
+description: You can set up Dagster to automatically materialize assets when criteria are met.
+---
+
+# Auto-Materializing Assets <Experimental />
+
+You can set up Dagster to automatically materialize assets when criteria are met. This enables a declarative approach to asset scheduling – instead of defining imperative workflows to materialize your assets, you just describe the conditions under which they should be materialized.
+
+At a high-level, there are three main factors that can be used to determine when an asset is auto-materialized:
+
+1. Whether upstream data has changed
+2. Whether code has changed, which is tracked using [code versions](/concepts/assets/software-defined-assets#asset-code-versions)
+3. Any <PyObject object="FreshnessPolicy" pluralize /> set on the asset or downstream assets
+
+Assets can be auto-materialized "eagerly" – i.e. immediately after data or code changes occur. Or they can be auto-materialized "lazily" – i.e. by waiting until downstream <PyObject object="FreshnessPolicy" pluralize /> dictate that they need to be fresh. Or a mixture of both.
+
+Auto-materializing lazily is especially useful when dealing with source data that is constantly changing. As a batch system, Dagster
+
+<!-- TODO: some nice diagram that shows freshness policies making an impact backward and upstream changes making an impact forward? -->
+
+### Auto-materialize policies
+
+You can set up an asset to be auto-materialized by assigning it an <PyObject object="AutoMaterializePolicy" />. In this example, we use <PyObject object="AutoMaterializePolicy" method="eager" /> to indicate that, any time that `asset1` is materialized, `asset2` should be automatically materialized right after:
+
+```python file=/concepts/assets/auto_materialize_eager.py
+from dagster import AutoMaterializePolicy, asset
+
+
+@asset
+def asset1():
+    ...
+
+
+@asset(auto_materialize_policy=AutoMaterializePolicy.eager())
+def asset2(asset1):
+    ...
+```
+
+Note that the [Dagster Daemon](/some/link) needs to be running for auto-materialization to function. If you use the `dagster dev` command to run Dagster, the daemon will automatically run as part of it.
```

Comment:
This daemon is pretty different from sensors and schedules which default to off and require a UI change to turn on. I think it is worth calling out that these policies will take effect _as soon as you start dagster_ and perhaps explain how you would pause them (if thats possible? I guess in prod just kill that daemon or in OSS kill dagster or push a change to the code... seems scary that we don't have a way to quickly shut down a system that is going crazy without a code push)

Reply:
I'll update the language to point out that it takes effect as soon as your start Dagster.

We're going to add a way to pausing them, but it's not in the codebase yet.